R/compare_timeseries.R

```diff
@@ -64,30 +66,56 @@ compare_timeseries <- function(obs_rts = NULL,
                                serial_interval = NULL,
                                rdist = NULL) {
 
+  ## Make a nested tibble of timseries and observed ata
+  data_tibble <- tibble::tibble(
+                                 timeseries = unique(obs_cases$timeseries),
+                                 rts = obs_rts %>%
+                                   dplyr::group_split(timeseries),
+                                 cases = obs_cases %>%
+                                   dplyr::group_split(timeseries)
+                                 )
 
-  obs_rts <- obs_rts %>%
-    dplyr::group_split(timeseries)
+  ## Make a tibble of all data and model combinations
+  combinations <- tidyr::expand_grid(data = list(data_tibble), models = models) %>%
+    tidyr::unnest("data") %>%
+    dplyr::mutate(model_name = names(models))
 
-  timeseries <- unique(obs_cases$timeseries)
+  ## Safe model evaluation
+  safe_eval <- purrr::safely(evaluate_model)
 
-  obs_cases <- obs_cases %>%
-    dplyr::group_split(timeseries)
 
-  evaluations <-
-    furrr::future_map2(obs_rts, obs_cases,
-                       ~ compare_models(obs_rts = dplyr::select(.x, -timeseries),
-                                        obs_cases = dplyr::select(.y, -timeseries),
-                                        models = models,
-                                        horizon = horizon , samples = samples,
-                                        bound_rt = bound_rt, timeout = timeout,
-                                        serial_interval = serial_interval,
-                                        min_points = min_points,
-                                        rdist = rdist),
-                       .progress = TRUE) %>%
-    setNames(timeseries) %>%
-    purrr::transpose() %>%
-    purrr::map(~ dplyr::bind_rows(., .id = "timeseries"))
+  ## Run each row combination and pull out the model evaluation
+  evaluations <- combinations %>%
+    dplyr::mutate(
+      eval = furrr::future_pmap(list(rts, cases, models),
+                                function(rt, case, model) {
+                                  safe_eval(
+                                    obs_rts = rt,
+                                    obs_cases = case,
+                                    model = model,
+                                    horizon = horizon,
+                                    samples = samples,
+                                    bound_rt = bound_rt,
+                                    timeout = timeout,
+                                    serial_interval = serial_interval,
+                                    min_points = min_points,
+                                    rdist = rdist
+                                  )[[1]]}),
```

Comment:
for readability it might be good to call the element with its name and not with [[1]]

Reply:
This is a `safely` element and so not meaninful